### DISCUSSION TITLE
Translating source locations to locations in the AST

### DISCUSSION THREAD

**Discussion by KennethDRoe** [MikePopoloski/slang#999] [Q&A]
I would like to modify the printJson function so that each AST node is printed with a source range showing the beginning and end of the object.  I see an ASTSerializer class that is used to print out json objects.  It has a class member ASTSerializer::visit which has the code for printing the objects.  What do I need to add to this code to print the location of the object in the source?

**Comment by MikePopoloski** [comment]
Each symbol has a `location` member. Also most AST constructs store a `syntax` pointer from which can be obtained a source range.

**Comment by KennethDRoe** [comment]
I've added the necessary changes to print out source line information in JSon files in a local fork here: https://github.com/KennethDRoe/slang.git.  The changes are all in ASTSerializer.  Is this class used for anything other than printing json files?  I saw a some regression failures after making the changes.  Are there known failures in master?

**Reply by MikePopoloski** [reply]
ASTSerializer is only used for serializing JSON (you should be able to find all uses with a simple grep of the codebase). You can see that there are no test failures in master by looking at the GitHub actions tab for the project.

**Comment by kendroe** [comment]
It looks like I get the same failures with a clean tree without my changes.  I suspect the issue is with my environment.  I'm building on an M1 Mac with Xcode 15.2 (15C500b), cmake 3.27.5 and Python 3.11.9.  I looked at your workflows.  It looks like you don't test on the Mac.

**Reply by MikePopoloski** [reply]
The CI build runs the tests on macos-13, which is probably not running Apple Silicon.

**Comment by kendroe** [comment]
Can you add me as a collaborator?  I'd like to submit a pull request for the change I made to print source file information in the json dump.

**Reply by MikePopoloski** [reply]
You should be able to submit a pull request without me doing anything on my end.

**Comment by KennethDRoe** [comment]
I've updated master in my fork with the changes you requested.  The three
JSON tests are passing.  I don't have a way of running the entire test
suite on one of your platforms.  The same tests are failing on my M1 Mac as
before.  Let me know what you think.

**Reply by MikePopoloski** [reply]
Ok, I'll take a look.

It would be nice if the tests passed on Apple Silicon as well but I have no way of running it myself. Unrelated to your diff, but is that something you could look into? AFAIK the tests pass fine on ARM Linux and x86 macOS, so it's a surprise to me that they don't also pass on ARM macOS.
